pkgs/by-name/gh/ghostty/package.nix

```
@@ -0,0 +1,139 @@
+  gtk4,
+  harfbuzz,
+  libadwaita,
+  libGL,
+  libpng,
+  libX11,
+  libXcursor,
+  libXi,
+  libXrandr,
+  ncurses,
+  nix-update-script,
+  oniguruma,
+  pandoc,
+  pkg-config,
+  wrapGAppsHook4,
+  zig_0_13,
+  zlib,
+}:
+let
+  # Ghostty needs to be built with --release=fast, --release=debug and
+  # --release=safe enable too many runtime safety checks.
+  zig_hook = zig_0_13.hook.overrideAttrs {
+    zig_default_flags = "-Dcpu=baseline -Doptimize=ReleaseFast --color off";
+  };
+in
+stdenv.mkDerivation (final: {
+  pname = "ghostty";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "ghostty-org";
+    repo = "ghostty";
+    tag = "v${final.version}";
+    hash = "sha256-AHI1Z4mfgXkNwQA8xYq4tS0/BARbHL7gQUT41vCxQTM=";
+  };
+
+  nativeBuildInputs = [
+    git
+    ncurses
+    pandoc
+    pkg-config
+    wrapGAppsHook4
+    zig_hook
+  ];
+
+  buildInputs = [
+    bzip2
+    expat
+    fontconfig
+    freetype
+    glib
+    gsettings-desktop-schemas
+    gtk4
+    harfbuzz
+    libadwaita
+    libGL
+    libpng
+    libX11
+    libXcursor
+    libXi
+    libXrandr
+    oniguruma
+    zlib
+  ];
+
+  dontPatch = true;
+  dontConfigure = true;
+  # doCheck is set to false because unit tests currently fail inside the Nix sandbox.
+  doCheck = false;
+
+  deps = callPackage ./deps.nix { name = "${final.pname}-cache-${final.version}"; };
+
+  zigBuildFlags = [
+    "--system"
+    "${final.deps}"
+    "-Dversion-string=${final.version}"
+  ];
+
+  zigCheckFlags = final.zigBuildFlags;
+
+  outputs = [
+    "out"
+    "terminfo"
+    "shell_integration"
+    "vim"
+  ];
+
+  postInstall = ''
+    mkdir -p "$terminfo/share"
+    mv "$out/share/terminfo" "$terminfo/share/terminfo"
+    ln -sf "$terminfo/share/terminfo" "$out/share/terminfo"
+
+    mkdir -p "$shell_integration"
+    mv "$out/share/ghostty/shell-integration" "$shell_integration/shell-integration"
+    ln -sf "$shell_integration/shell-integration" "$out/share/ghostty/shell-integration"
+
+    mv "$out/share/vim/vimfiles" "$vim"
+    ln -sf "$vim" "$out/share/vim/vimfiles"
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(--prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libX11 ]}")
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    homepage = "https://ghostty.org/";
+    description = "Fast, native, feature-rich terminal emulator pushing modern features";
+    longDescription = ''
+      Ghostty is a terminal emulator that differentiates itself by being
+      fast, feature-rich, and native. While there are many excellent terminal
+      emulators available, they all force you to choose between speed,
+      features, or native UIs. Ghostty provides all three.
+    '';
+    downloadPage = "https://ghostty.org/download";
+
+    license = lib.licenses.mit;
+    platforms = lib.platforms.linux;
+    mainProgram = "ghostty";
+    outputsToInstall = final.outputs;
+    maintainers = with lib.maintainers; [
+      jcollie
+      pluiedev
```

```suggestion
      pluiedev
      getchoo
```
Feel free to add me as well. I'd love to help out on Mac support and exposing some more build configuration options in the future :)